keps/sig-auth/2784-csr-duration/README.md

```diff
@@ -0,0 +1,473 @@
+simplicity of implementation and low risk nature of the proposal, the alpha stage
+and related feature gate have been omitted from this design.
+
+Clients that do not set the `spec.expirationSeconds` field will observe no change
+in behavior, even during upgrades and downgrades.
+
+### Version Skew Strategy
+
+There are three actors we need to consider:
+
+1. API Server
+2. Controller manager
+3. Clients that create CSRs
+
+As noted above, old clients observe no change in behavior, thus we assume for the
+discussion below that all clients have been upgraded and are attempting to set
+the `spec.expirationSeconds` field.
+
+Once the API server is upgraded, clients will be able to set the new field.  For
+the purpose of this design, upgrading other components before the API server is
+of no consequence as it is impossible to set the new field without the API server
+knowing of its existence.
+
+#### Scenario 1
+
+1. Upgraded API server
+2. Not upgraded (or partially upgraded) controller manager
+3. Upgraded client
+
+In this scenario, the requested `spec.expirationSeconds` may be ignored because
+the controller manger will not understand this field.  This is harmless and
+represents the status quo.
+
+#### Scenario 2
+
+1. Partially upgraded API server
+2. Upgraded controller manager
+3. Upgraded client
+
+In this scenario, the requested `spec.expirationSeconds` may be ignored because
+old API servers will silently drop the field on update.  This is harmless
+represents and the status quo.
+
+The CSR API is resilient to split brain scenarios as unknown fields are silently
+dropped and the `spec` fields are immutable after creation [1] [2] [3].
+
+[1]: https://github.com/kubernetes/kubernetes/blob/24b716673caae31f070b06a337bc12c97ff1d4cb/pkg/registry/certificates/certificates/strategy.go#L104-L112
+[2]: https://github.com/kubernetes/kubernetes/blob/24b716673caae31f070b06a337bc12c97ff1d4cb/pkg/registry/certificates/certificates/strategy.go#L175-L176
+[3]: https://github.com/kubernetes/kubernetes/blob/24b716673caae31f070b06a337bc12c97ff1d4cb/pkg/registry/certificates/certificates/strategy.go#L297-L298
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- Other
+  - Describe the mechanism:
+      As written above, this functionality will be enabled by default with no configuration options.
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+      N/A
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+      N/A
+
+###### Does enabling the feature change any default behavior?
+
+Existing clients would continue to leave `spec.expirationSeconds` unset and thus
+would observe no difference in behavior.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+N/A
```

Comment:
reference the feature gate... if use of the shorter expirations reveals a problem, controller-manager should be able to turn off honoring the field using a gate

Reply:
So a feature gate that:

1. Is enabled by default
2. Starts out at beta

?

Reply:
> So a feature gate that:
> 
> 1. Is enabled by default
> 2. Starts out at beta

Yes.  That gives us a way to turn it off in a running cluster without rolling back.

Reply:
Added explicit feature gate.